_scala3-reference/contextual/derivation.md

```diff
@@ -218,7 +218,7 @@ instance for the appropriate ADT subtype using the auxiliary method `check` (4).
 ```scala
 import scala.deriving.Mirror
 
-def eqSum[T](s: Mirror.SumOf[T], elems: List[Eq[_]]): Eq[T] =
+def eqSum[T](s: Mirror.SumOf[T], elems: => List[Eq[_]]): Eq[T] =
```

Comment:
This does not look like a good idea, each time we would do `check(elems(ordx))(x, y)` we would need to recompute the full list.